app/assets/javascripts/discourse/app/components/d-modal.gjs

```diff
@@ -77,9 +121,25 @@ export default class DModal extends Component {
       return;
     }
 
-    return this.args.closeModal?.({
-      initiatedBy: CLOSE_INITIATED_BY_CLICK_OUTSIDE,
-    });
+    return this.closeModal(CLOSE_INITIATED_BY_CLICK_OUTSIDE);
+  }
+
+  @action
+  closeModal(initiatedBy) {
+    if (!this.args.closeModal) {
+      return;
+    }
+
+    this.closing = true;
+
+    this.wrapperElement.animate([{ transform: `translateY(100%)` }], {
+      duration: CLOSE_MODAL_DURATION,
+      fill: "forwards",
+    }).finished;
+
+    this.closeModalHandler = later(() => {
+      this.args.closeModal({ initiatedBy });
+    }, CLOSE_MODAL_DURATION);
```

Comment:
Could we use the animation promise for this?

```suggestion
    await this.wrapperElement.animate([{ transform: `translateY(100%)` }], {
      duration: CLOSE_MODAL_DURATION,
      fill: "forwards",
    }).finished;

    this.args.closeModal({ initiatedBy });
```

Reply:
Ah yes good catch indeed we don't need the later anymore here